CONTRIBUTING.md

```diff
@@ -0,0 +1,193 @@
+# Programming Guidelines
+
+## General Guidelines
+
+### Why do we even need guidelines in the first place?
+
+At the beginning, Space Concordia Robotics was small. The code base was maintained by two individuals that knew the code very well. The robotics team is now growing with new members of a wide range of programming experience. Due to the negligence of the best practices, the code became spaghetti and almost impossible to read for new members. It is hard to debug and to test. We need clean code because
+
+#### Clean code is easier to read
+
+#### Clean code is easier to debug and is more maintainable
+
+#### Well designed code is scalable
+
+#### Well planned code is fun to build on
+
+#### There is a cost to having a total mess
+Teams that have moved swiftly at the beginning of projects can find themselves moving at a much slower rate further in. The term [Technical Debt](https://en.wikipedia.org/wiki/Technical_debt) describes the cost of the additional work that needs to be done (because of slower debugging, more bugs, etc) due to messy code.
+
+## Best Practices
+
+These best practices are mostly taken from the book `Clean Code, A Handbook of Agile Craftsmanship` by Robert C. Martin 
+
+### 1. Functions
+
+#### 1.1 Do one thing
+
+Functions should do one thing. They should do it well. They should do it only. One symptom of doing more than one thing is having sections within functions. These can oftentimes easily be split up into subroutines that will only have a specific role.
+
+#### 1.2 Size
+
+Size does matter! The size of the function should be **small**. A function that cannot be read in one screen should definitely be split up.
```

Comment:
What if it is my phone screen? I think it's a little too open to interpretation

Reply:
So I don't want to put a hard limit to the number of lines of code. I updated the paragraph with information from this https://softwareengineering.stackexchange.com/a/133406/255377